webapp/src/pages/ChatRoom/MessageInput.css

```diff
@@ -0,0 +1,72 @@
+.message-input-form {
+  width: 100%;
+  display: grid;
+  gap: 0.5rem;
+  grid-template-columns: 1fr;
+  grid-template-rows: 1fr auto;
+  grid-template-areas:
+    'message-input'
+    'submit-button';
+}
+
+.message-input-grow-wrap {
+  grid-area: message-input;
+  /* easy way to plop the elements on top of each other and have them both sized based on the tallest one's height */
+  display: grid;
+}
+
+.message-input-grow-wrap::after {
+  /* Note the weird space! Needed to preventy jumpy behavior */
+  content: attr(data-replicated-value) ' ';
+
+  /* This is how textarea text behaves */
+  white-space: pre-wrap;
+
+  /* Hidden from view, clicks, and screen readers */
+  visibility: hidden;
+}
+
+.message-input-text {
+  /* You could leave this, but after a user resizes, then it ruins the auto sizing */
+  resize: none;
+}
+
+.message-input-text::placeholder {
+  color: var(--color-light);
+  opacity: 1;
+}
+
+.message-input-grow-wrap::after,
+.message-input-text {
+  /* Identical styling required!! */
+  outline: none;
+  border: none;
+  margin: 0;
+  min-height: 3rem;
+  max-height: 12rem;
+  word-break: break-all;
```

Comment:
why do we have here `break-all` ?
It breaks lines in the middle of words. I have never seen any text field where this is normal or desired behaviour.

Reply:
I've added this before when we had a `contenteditable` div because very long words broke the layout.
I see it's not the case anymore, I will remove it :+1: